File: assign4/javascript/src/guessingGameConsole.js
const { playGuessingGame, generateRandomNumber, Comparison } = require('./guessingGame');
const prompt = require('prompt-sync')();

const readGuess = () => parseInt(prompt("Please enter your guess (a number between 1 and 100): "), 10);

const displayResult = (result) => console.log({
  [Comparison.HIGH]: "Your guess is high!",
  [Comparison.LOW]: "Your guess is low!",
  [Comparison.EQUAL]: "Congratulations! You guessed the correct number!"
}[result]);

function main() {
  const target = generateRandomNumber(Date.now().toString());
  
  console.log("Welcome to the Guessing Game!\nTry to guess the number between 1 and 100!");
  
  const attempts = playGuessingGame(target, readGuess, displayResult);
  
  console.log(`You guessed the number in ${attempts} attempts.`);
}

main();

File: assign4/rust/src/guessing_game_console.rs
use guessing_game::{play_guessing_game, generate_random_number, Comparison};
use std::io::{self, Write};
use std::time::{SystemTime, UNIX_EPOCH};

fn main() { 

  let target = generate_random_number(SystemTime::now().duration_since(UNIX_EPOCH).unwrap().as_secs()); 

  println!("Welcome to the Guessing Game!");
  println!("I'm thinking of a number between 1 and 100. Try to guess it!");

  let read_guess = || {
    print!("Please enter your guess (a number between 1 and 100): ");
    io::stdout().flush().unwrap();
    io::stdin().lines().next().unwrap().unwrap().trim().parse().unwrap()
  };

  let display = |result: Comparison| match result {
    Comparison::HIGH => println!("Your guess is high!"),
    Comparison::LOW => println!("Your guess is low!"),
    Comparison::EQUAL => println!("Congratulations! You guessed the correct number!"),
  };

  let attempts = play_guessing_game(target, read_guess, display);
  
  println!("You guessed the number in {} attempts.", attempts);
}
